src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -1467,6 +1467,135 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id,
+      IMAGE_CEE_CS_CALLCONV_DEFAULT,
+      0,
+      ELEMENT_TYPE_BOOLEAN,
+      ELEMENT_TYPE_OBJECT
+  };
+  mdMethodDef alreadyLoadedMethodToken;
+  hr = metadata_emit->DefineMethod(
+      new_type_def, "IsAlreadyLoaded"_W.c_str(), mdStatic | mdPrivate,
+      already_loaded_signature, sizeof(already_loaded_signature), 0, 0,
+      &alreadyLoadedMethodToken);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineMethod IsAlreadyLoaded failed");
+    return hr;
+  }
+
+  // Get a TypeRef for System.Threading.Interlocked
+  mdTypeRef interlocked_type_ref;
+  hr = metadata_emit->DefineTypeRefByName(mscorlib_ref, "System.Threading.Interlocked"_W.c_str(), &interlocked_type_ref);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineTypeRefByName interlocked_type_ref failed");
+    return hr;
+  }
+
+  // Create method signature for System.Threading.Interlocked::CompareExchange(int32&, int32, int32)
+  COR_SIGNATURE interlocked_compare_exchange_signature[] = {
+      IMAGE_CEE_CS_CALLCONV_DEFAULT,
+      3,
+      ELEMENT_TYPE_I4,
+      ELEMENT_TYPE_BYREF,
+      ELEMENT_TYPE_I4,
+      ELEMENT_TYPE_I4,
+      ELEMENT_TYPE_I4
+  };
+
+  mdMemberRef interlocked_compare_member_ref;
+  hr = metadata_emit->DefineMemberRef(
+      interlocked_type_ref, "CompareExchange"_W.c_str(),
+      interlocked_compare_exchange_signature,
+      sizeof(interlocked_compare_exchange_signature),
+      &interlocked_compare_member_ref);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineMemberRef CompareExchange failed");
+    return hr;
+  }
+
+  /////////////////////////////////////////////
+  // Add IL instructions into the IsAlreadyLoaded method
+  //
+  //  static int _isAssemblyLoaded = 0;
+  //  
+  //  public static bool IsAlreadyLoaded() {
+  //      return Interlocked.CompareExchange(ref _isAssemblyLoaded, 1, 0) == 1;
+  //  }
+  //
+  ILRewriter rewriter_already_loaded(this->info_, nullptr, module_id, alreadyLoadedMethodToken);
+  rewriter_already_loaded.InitializeTiny();
+
+  ILInstr* pALFirstInstr = rewriter_already_loaded.GetILList()->m_pNext;
+  ILInstr* pALNewInstr = NULL;
+
+  // ldsflda _isAssemblyLoaded : Load the address of the "_isAssemblyLoaded" static var
+  pALNewInstr = rewriter_already_loaded.NewILInstr();
+  pALNewInstr->m_opcode = CEE_LDSFLDA;
+  pALNewInstr->m_Arg32 = isAssemblyLoadedFieldToken;
+  rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+  
+  // ldc.i4.1 : Load the constant 1 (int) to the stack
+  pALNewInstr = rewriter_already_loaded.NewILInstr();
+  pALNewInstr->m_opcode = CEE_LDC_I4_1;
+  rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+  // ldc.i4.0 : Load the constant 0 (int) to the stack
+  pALNewInstr = rewriter_already_loaded.NewILInstr();
+  pALNewInstr->m_opcode = CEE_LDC_I4_0;
+  rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+  // call System.AppDomain System.AppDomain.CurrentDomain property
```

Comment:
I think you mean Interlocked.CompareExchange :D

Reply:
ups, you're right

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1055/commits/2c168c30f743bbfcbdc44b8c75d27e37195053c2